core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -3797,6 +3797,29 @@ public static List distinct(List list) {
     return new ArrayList<>(result);
   }
 
+  /** Support the ARRAY_EXCEPT function. */
```

Comment:
Is the result order of these functions consistents with the parametes order? For example:
> SELECT array_union(array(1, 2, 3), array(1, 3, 5));
> return: [1,2,3,5]

Is it valid to return [5,1,3,2]?

Reply:
If we don't consider order here, can we use `HashSet` to implement to get better performance?
I see spark's implementation, it uses `OpenHashSet` which is not ordered from what I know so far.

Reply:
@herunkang2018 the spark supports it has two way because the runtime binary compare
1) complex type, compare using for loop
2) simple type using OpenHashSet, and it just for check
and i supports array_union in flink , you can see here https://github.com/apache/flink/pull/22483#event-9255748646

the result should consider order, which is different from multiset_union/multiset_intersect in calcite implementation

Reply:
I see.